qa/integration/specs/pipeline_log_spec.rb

```diff
@@ -44,4 +45,35 @@
     expect(File.exists?(plainlog_file)).to be true
     expect(IO.read(plainlog_file) =~ /\[logstash.javapipeline\s*\]\[#{pipeline_name}\]/).to be > 0
   end
+
+  it "write pipeline config in logs - source:config string" do
+    pipeline_name = "custom_pipeline"
+    settings = {
+      "path.logs" => temp_dir,
+      "pipeline.id" => pipeline_name
+    }
+    IO.write(@ls.application_settings_file, settings.to_yaml)
+    @ls.spawn_logstash("-w", "1" , "-e", config)
+    @ls.wait_for_logstash
+    sleep 2 until @ls.exited?
+    plainlog_file = "#{temp_dir}/logstash-plain.log"
+    expect(File.exists?(plainlog_file)).to be true
+    expect(IO.read(plainlog_file) =~ /Starting pipeline.*"pipeline.sources"=>\["config string"\]/).to be > 0
+  end
+
+  it "write pipeline config in logs - source:config file" do
+    pipeline_name = "custom_pipeline"
+    settings = {
+      "path.logs" => temp_dir,
+      "pipeline.id" => pipeline_name
+    }
+    IO.write(@ls.application_settings_file, settings.to_yaml)
+    puts "initial_config_file: #{initial_config_file}"
+    @ls.spawn_logstash("-w", "1", "-f", "#{initial_config_file}")
+    @ls.wait_for_logstash
```

Comment:
I _think_ this line might be causing the build flakiness. as Logstash comes up and down before the call to the API endpoint can be completed. Removing this line from this (and the other logging tests) might help

Reply:
@robbavey fixed this bringing code to check LS process termination